### PR TITLE
Fix shellcheck failures in test/images/volume/.....

### PR DESCRIPTION
**Pull request by xichengliudui**
**What type of PR is this?**

 /kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Ref #72956

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/images/volume/nfs/run_nfs.sh [hunk 1]
```diff
@@ -22,21 +22,22 @@ function start()
     while getopts "G:" opt; do
         case ${opt} in
             G) gid=${OPTARG};;
-        esac
+            *) echo "accept: -G gid = ${OPTARG}"
```
Comment (xichengliudui) [inline test/images/volume/nfs/run_nfs.sh]: Why did it change? What did it fix?
```
In ./test/images/volume/nfs/run_nfs.sh line 23:
        case ${opt} in
        ^-- SC2220: Invalid flags are not handled. Add a *) case.
Ref: https://github.com/koalaman/shellcheck/wiki/SC2220
```
example: 
**Problematic code**:
```
#!/bin/sh
while getopts "vr" f
do
  case "$f" in
    v) echo "verbose" ;;
    r) echo "recursive" ;;
  esac
done
```
**Correct code:**:
```
#!/bin/sh
while getopts "vr" f
do
  case "$f" in
    v) echo "verbose" ;;
    r) echo "recursive" ;;
    *) echo "usage: $0 [-v] [-r]" >&2
       exit 1 ;;
  esac
done
```

File: test/images/volume/nfs/run_nfs.sh [hunk 2]
```diff
@@ -22,21 +22,22 @@ function start()
...
+	esac
     done
-    shift $(($OPTIND - 1))
+    shift $((OPTIND - 1))
```
Comment (xichengliudui) [inline test/images/volume/nfs/run_nfs.sh]: Why did it change? What did it fix?
```
In ./test/images/volume/nfs/run_nfs.sh line 27:
    shift $(($OPTIND - 1))
             ^-----^ SC2004: $/${} is unnecessary on arithmetic variables
Ref: https://github.com/koalaman/shellcheck/wiki/SC2004
```
example:
**Problematic code**:
```
echo $(($n + ${arr[i]}))
```
**Correct code**: 
```
echo $((n + arr[i]))
```

**Comment by SataQiu** [comment]
thanks @xichengliudui 
/kind cleanup
/sig testing
/priority important-soon

File: test/images/volume/nfs/run_nfs.sh (hunk 1 above)
Comment (BenTheElder) [reply test/images/volume/nfs/run_nfs.sh]: this is not correct, it should look more like the suggested code with:
- needs to either return or exit with an error depending on the context
- usage string should be the program usage and go to stderr

Comment (xichengliudui) [reply test/images/volume/nfs/run_nfs.sh]: Thanks for the comments, which I'll update later.

**Pull request closed without merging**